assets/js/components/ResetButton.test.js

```diff
@@ -0,0 +1,109 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { render, createTestRegistry, fireEvent, act } from '../../../tests/js/test-utils';
+import { STORE_NAME as CORE_SITE } from '../googlesitekit/datastore/site/constants';
+import ResetButton from './ResetButton';
+import { subscribeUntil } from '../../../tests/js/utils';
+
+describe( 'ResetButton', () => {
+	let registry;
+	const adminURL = 'http://example.com/';
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		registry.dispatch( CORE_SITE ).receiveSiteInfo( { adminURL } );
+	} );
+
+	it( 'should render "Reset Site Kit" as the default child', () => {
+		const { container } = render( <ResetButton />, { registry } );
+		expect( container ).toHaveTextContent( 'Reset Site Kit' );
+	} );
+
+	it( 'should render passed children inside of the button', () => {
+		const { container } = render( <ResetButton>Test Value</ResetButton>, { registry } );
+		expect( container ).toHaveTextContent( 'Test Value' );
+		expect( document.querySelector( '.mdc-dialog--open' ) ).toBeNull();
+	} );
+
+	describe( 'after click', () => {
+		let container;
+		beforeEach( () => {
+			( { container } = render( <ResetButton />, { registry } ) );
+			fireEvent.click( container.querySelector( '.googlesitekit-reset-button' ) );
+		} );
+
+		it( 'should open the dialog', async () => {
+			expect( document.querySelector( '.mdc-dialog--open' ) ).toBeTruthy();
+		} );
+
+		it( 'should show reset and cancel buttons', async () => {
+			expect( document.querySelector( '.mdc-dialog--open .mdc-dialog__cancel-button' ) ).toBeTruthy();
+			expect( document.querySelector( '.mdc-dialog--open .mdc-button--danger' ) ).toBeTruthy();
+		} );
+
+		it( 'should close the modal on clicking cancel', async () => {
+			fireEvent.click( document.querySelector( '.mdc-dialog--open .mdc-dialog__cancel-button' ) );
+			expect( document.querySelector( '.mdc-dialog--open' ) ).toBeNull();
+		} );
+
+		it( 'should close the modal on pressing escape key', async () => {
+			fireEvent.keyUp( global, { keyCode: 27 } );
+			expect( document.querySelector( '.mdc-dialog--open' ) ).toBeNull();
+		} );
+
+		it( 'should close the modal on clicking Reset and ', async () => {
+			const response = true;
+			fetchMock.postOnce(
+				/^\/google-site-kit\/v1\/core\/site\/data\/reset/,
+				{ body: JSON.stringify( response ), status: 200 },
+			);
+
+			delete global.location;
+			global.location = {
+				href: '',
+			};
+
+			const localStorageMock = ( function() {
+				let store = { randomKey: 'randomValue' };
+				return {
+					getItem( key ) {
+						return store[ key ];
+					},
+					clear() {
+						store = {};
+					},
+				};
+			}() );
+			Object.defineProperty( global, 'localStorage', { value: localStorageMock } );
```

Comment:
We include `jest-localstorage-mock` in our tests already, so I don't think you need to do this here 🙂 

See: https://github.com/google/site-kit-wp/blob/5e51067e3a4bc93f7cd5419d798f6ee2d0f56713/assets/js/googlesitekit/api/cache.test.js#L123 for an example of it in-use.